src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,126 @@
+package duke;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Class handling interactions with the users
+ */
+public class Ui {
+    private Scanner s = new Scanner(System.in);
+
+    /**
+     * Retrieve the next command from the user and executes it.
+     *
+     * @return The next line of input
+     */
+    public String nextLine() {
+        return s.nextLine();
+    }
+
+    /**
+     * Prints the welcome message.
+     */
+    public void printWelcome() {
+        System.out.println("Hello! I'm \nWhat can I do for you?");
+    }
+
+    /**
+     * Prints the goodbye message.
+     */
+    public void printGoodbye() {
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+
+    /**
+     * Prints the message for successful load.
+     */
+    public void printLoadSuccess() {
+        System.out.println("Save file successfully loaded");
+    }
+
+    /**
+     * Prints the list of task.
+     *
+     * @param tasks TaskList to be printed.
+     */
+    public void printList(TaskList tasks) {
+        System.out.println("---------");
+        System.out.println(tasks.toString());
+        System.out.println("---------");
+    }
+
+
+    /**
+     * Prints the error of a given DukeException.
+     *
+     * @param err Exception to be printed.
+     */
+    public void printDukeException(DukeException err) {
+        System.out.println("---------");
+        System.out.println("Error: " + err.getMessage());
+        System.out.println("---------");
+    }
+
+
+    /**
+     * Print message when adding a new task.
+     * @param tasks TaskList of the program.
+     * @throws DukeException When TaskList cannot retreive the last element in the list.
+     */
+    public void printNewTask(TaskList tasks) throws DukeException {
+        System.out.println("---------");
```

Comment:
I think this "-----", "-----", can be abstracted as a method, e.g. formatMessage()